src/API/Controllers/CustomersController.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using AppServices;
+using DomainModels;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class CustomersController : ControllerBase
+{
+    private readonly ICustomerAppService _appService;
+
+    public CustomersController(ICustomerAppService appService)
+    {
+        _appService = appService;
+    }
+
+    [HttpPost]
+    public IActionResult Post(Customer customer)
+    {
+        var createdCustomer = _appService.CreateCustomer(customer); 
+        return createdCustomer.isValid
+            ? Created("~http://localhost:5160/api/Customers", createdCustomer.message)
+            : BadRequest(createdCustomer.message);      
+    }
+
+    [HttpGet]
+    public IActionResult Get()
+    {
+        var CustomersFound = _appService.GetCustomers();
+        return Ok(CustomersFound);
+    }
+
+    [HttpGet("byId/{id}")]
+    public IActionResult GetById(Guid id)
+    {
+        var CustomerFoundId = _appService.GetCustomerById(id);
+        return CustomerFoundId is null
+            ? NotFound($"Customer para o id [{id}] não foi encontrado.")
+            : Ok(CustomerFoundId);
+    }
+
+    [HttpGet("byName")]
+    public IActionResult GetByName(string fullName)
```

Comment:
Endpoint já é Get, seria redundante ter o by
```suggestion
    [HttpGet("name")]
    public IActionResult GetByName(string fullName)
```